File: v1-java-vs-ruby_r/FigurenTest.rb
require "./Figur"
require "./Punkt"
require "./Kreis"
require "./Rechteck"
require "./Ueberlagert"

class FigurenTest
  
  r1 = Rechteck.new(Punkt.new(3,3))
  r2 = Rechteck.new(Punkt.new(2.5,2.5))
  r3 = Rechteck.new(Punkt.new(3.5,3.5))
  k1 = Kreis.new(Punkt.new(3.5,4))
  k2 = Kreis.new(Punkt.new(2,2),3)
  k3 = Kreis.new(Punkt.new(3,4),1.5)
  u1 = Ueberlagert.new(r1,r2)
  u2 = Ueberlagert.new(r3,k1)
  u3 = Ueberlagert.new(k2,k3)
  u4 = Ueberlagert.new(u1,u2)
  u5 = Ueberlagert.new(u3,u4)
  p u4
  
  punkt = Punkt.new(3,4);
  p u5.in(punkt)
  puts  u4.in(punkt)
  p u3.in(4)  # zeigt, dass die Typprüfung nur zur Laufzeit stattfindet
  p u3.in(punkt)
  p u2.in(punkt)
  p u1.in(punkt)
  p k2.in(punkt)
  p k1.in(punkt)
  p r3.in(punkt)
  p r2.in(punkt)
  p r1.in(punkt)

end

File: v1-java-vs-ruby/src/figuren/FigurenTest.java
package figuren;

import original.Punkt;

public class FigurenTest {

    public static void main(String[] args) {
        Rechteck r1;
        Figur k2;
        Kreis k3;
        Figur f1;
        String s="12";

        alternative.Punkt ap = new alternative.Punkt(4,7);


        k2 = new Kreis(new Punkt(4, 4));
        r1 = new Rechteck(new Punkt(3, 3));

        k3 = new Kreis(new Punkt(1,1),4);

        f1 = new Rechteck(new Punkt(2, 2));
        f1 = new Kreis(new Punkt(1,1),4);

        s.concat("34");

      // k2.radius = 15;

        r1.in(new Punkt(1, 9));
        k2.in(new Punkt(1, 9));

        System.out.println(k2);

        System.out.println(r1);

        Punkt punkt = new Punkt(3, 4);
        System.out.println(r1.in(punkt));

        String o = readme();

        int[] iary = new int[10];

        iary[9] = 45;
    }

    public static String readme() {
        System.out.println("Readme");
        return "Readme";
    }
}